adapter/internal/messaging/azure_listener.go

```diff
@@ -22,13 +22,30 @@ import (
 	"github.com/wso2/product-microgateway/adapter/config"
 	logger "github.com/wso2/product-microgateway/adapter/internal/loggers"
 	msg "github.com/wso2/product-microgateway/adapter/pkg/messaging"
+	servicebus "github.com/Azure/azure-service-bus-go"
+	"github.com/wso2/product-microgateway/adapter/pkg/health"
+	"time"
+)
+
+const (
+	componentName string = "adapter"
+	subscriptionIdleTimeDuration = time.Duration(72 * time.Hour)
 )
 
 // InitiateAndProcessEvents to pass event consumption
 func InitiateAndProcessEvents(config *config.Config) {
+	var err error
+	var namespace *servicebus.Namespace
 	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Starting InitiateAndProcessEvents method")
 	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] EventListeningEndpoint is ",
 		config.ControlPlane.ASBConnectionParameters.EventListeningEndpoint)
-	msg.InitiateBrokerConnection(config.ControlPlane.ASBConnectionParameters.EventListeningEndpoint)
-
+	namespace, availableTopicList, err := msg.InitiateBrokerConnectionAndGetAvailableTopics(
+		config.ControlPlane.ASBConnectionParameters.EventListeningEndpoint)
+	health.SetControlPlaneBrokerStatus(err == nil)
+	if err == nil {
+		logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Initiated broker connection successfully ")
+		msg.InitiateConsumers(namespace, availableTopicList, componentName, subscriptionIdleTimeDuration)
+	}
+	go handleAzureNotification()
```

Comment:
I suppose we shouldn't continue if there's any error?

Reply:
this is done